client.html

```diff
@@ -53,6 +78,221 @@
+                var elem = elems[i];
+                elem.parentNode.classList.add('branch-highlight-temp');
+            }
+        }
+
+        function highlightBranchTempClear(className) {
+            var elems = document.getElementsByClassName(className);
+            for (var i = 0; i < elems.length; i++) {
+                var elem = elems[i];
+                elem.parentNode.classList.remove('branch-highlight-temp');
+            }
+        }
+
+        function onClickBranchOrigin(ev) {
+            scrollIntoView(document.getElementById(ev.target.dataset.branchTarget), [ev.target]);
+            highlightBranchClearAll();
+            highlightBranchStay(ev.target.dataset.branchesClass);
+        }
+
+        function onClickBranchTarget(ev) {
+            // get all branch origins that branch to this branch target (ev.target)
+            var branchIndicatorsElems = document.getElementsByClassName(ev.target.dataset.branchesClass);
+            var branchOriginsElems = [];
+            for (var i = 0; i < branchIndicatorsElems.length; i++) {
+                if (branchIndicatorsElems[i] != ev.target) {
+                    branchOriginsElems.push(branchIndicatorsElems[i]);
+                }
+            }
+
+            // use a different origin as main scroll-to element each click
+            var scrollMainElemIdx;
+            if ('nextScrollMainElemIdx' in ev.target.dataset) {
+                scrollMainElemIdx = parseInt(ev.target.dataset.nextScrollMainElemIdx);
+            } else {
+                scrollMainElemIdx = 0;
+            }
+            scrollMainElemIdx %= branchOriginsElems.length;
+            ev.target.dataset.nextScrollMainElemIdx = scrollMainElemIdx + 1;
+
+            var scrollMainElem = branchOriginsElems[scrollMainElemIdx];
+            var scrollOtherElems = [ev.target];
+            for (var i = 0; i < branchOriginsElems.length; i++) {
+                if (i != scrollMainElemIdx) {
+                    scrollOtherElems.push(branchOriginsElems[i]);
+                }
+            }
+            scrollIntoView(scrollMainElem, scrollOtherElems);
+
+            highlightBranchClearAll();
+            highlightBranchStay(ev.target.dataset.branchesClass);
+        }
+
+        function onMouseEnterBranchIndicator(ev) {
+            highlightBranchTemp(ev.target.dataset.branchesClass,);
+        }
+
+        function onMouseLeaveBranchIndicator(ev) {
+            highlightBranchTempClear(ev.target.dataset.branchesClass);
+        }
+
+        function setDiffHtml(diffHtml) {
+            ASMDW.dom.diffContainer.innerHTML = diffHtml;
+            var tbody = ASMDW.dom.diffContainer.getElementsByTagName('tbody')[0];
+            for (var i = 0; i < tbody.children.length; i++) {
+                var tr = tbody.children[i];
+                for (var j = 0; j < 2 && j < tr.children.length; j++) {
+                    var td = tr.children[j];
+                    var spanChildren = td.getElementsByTagName('span');
```

Comment:
`diffContainer.querySelectorAll("tbody > tr > td > span")`

Reply:
how would it handle not going into the --threeway column? (in the loop I have `j<2`)
I didn't test it but I think we don't want to do anything about that column?

Reply:
Ah, that's true. Can add a `:not(:nth-child(3))` part to the selector